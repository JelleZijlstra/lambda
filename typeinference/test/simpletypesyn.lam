data either a b = Left a | Right b in
type with_err_code = either int in

let hd : with_err_code int = Right 3 in
let hd' : with_err_code string = Left 3 in
(hd, hd')