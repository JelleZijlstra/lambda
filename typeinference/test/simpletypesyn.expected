%%stdout%%
Result: module
	let __result__ = ((Right 3), (Left 3))
end
%%stderr%%
